checker/raw_result.go

```diff
@@ -111,6 +111,7 @@ const (
 // PinningDependenciesData represents pinned dependency data.
 type PinningDependenciesData struct {
 	Dependencies []Dependency
+	Incomplete   []error // jobs or files with errors may have incomplete results
```

Comment:
I left it as a generic since it doesn't always have to be an ElementError. For example, the error stored when parsing a Dockerfile is whatever kind of error is returned by `syntax.NewParser().Parse()`:

https://github.com/ossf/scorecard/pull/3515/files#diff-d861376c45f3cee9d7b3959c8bb1954fc1698ac3549a53ce369ce3a2e7c5962dR1062-R1067

ElementError is (currently) only used when the problem is parsing a workflow.

I didn't use `errors.New` or `sce.WithMessage` because I don't have full location information at the moment the problem arises, just the job name. I could have used those anyway, but I'd then have to parse the error string to extract the job name and recreate the error with the filename and step number.

I therefore thought it best to create an error type, especially since I think this approach (skipping problematic code and logging it in the details) can be applied in a few other cases and having an error type with location metadata might be useful.